### PR TITLE
[10.0][IMP]purchase_request

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
This PR means to incorporate to v10 the improvements made for v11:

- [x] Procurement groups satisfied by simple purchase requests. This closes #657 because this covers the case it solves.
- [x] Apply: https://github.com/OCA/purchase-workflow/pull/646

### CODE REVIEW

File: purchase_request/models/purchase_request.py [hunk 1]
```diff
@@ -115,6 +115,9 @@ def _track_subtype(self, init_values):
         compute='_compute_line_count',
         readonly=True
     )
+    product_id = fields.Many2one('product.product',
+                                 related='line_ids.product_id',
+                                 string='Product')
```
Comment (sergio-teruel) [inline purchase_request/models/purchase_request.py:121]: Add readonly for related fields

File: purchase_request_procurement/__manifest__.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
     'data': [
         "views/product_template.xml",
         "views/procurement_order.xml",
+        "views/purchase_request_view.xml",
     ],
     'demo': [],
```
Comment (sergio-teruel) [inline purchase_request_procurement/__manifest__.py]: remove empty keys

File: purchase_request_procurement/models/procurement_order.py [hunk 3]
```diff
@@ -27,29 +27,44 @@ def _prepare_purchase_request_line(self):
...
         self.ensure_one()
-
+        gpo = self.rule_id.group_propagation_option
+        group_id = (gpo == 'fixed' and self.group_id.id) or \
+                   (gpo == 'propagate' and self.group_id.id) or False
         return {
             'origin': self.origin,
             'company_id': self.company_id.id,
             'picking_type_id': self.rule_id.picking_type_id.id,
+            'group_id': group_id or False,
```
Comment (sergio-teruel) [inline purchase_request_procurement/models/procurement_order.py]: in this line (https://github.com/OCA/purchase-workflow/pull/659/files#diff-93d06fd5764b8ad47a588d9aec44e66aR39) group_id already is False so you can remove or False

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

File: purchase_request_procurement/10.0.2.0.1/post-migration.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# Copyright 2018 Eficent Business and IT Consulting Services S.L.
```
Comment (MiquelRForgeFlow) [inline purchase_request_procurement/10.0.2.0.1/post-migration.py]: The file should be in a folder named `10.0.1.2.1` instead

**Comment by AaronHForgeFlow** [comment]
Dear @sergio-teruel @mreficent 

Comments have been attended. An update in the review would be warmly appreciated.

Regards.

File: purchase_request_procurement/__manifest__.py [hunk 5]
```diff
@@ -4,7 +4,7 @@
 
 {
     'name': "Purchase Request Procurement",
-    'version': "10.0.1.0.1",
+    'version': "10.0.1.2.1",
```
Comment (MiquelRForgeFlow) [inline purchase_request_procurement/__manifest__.py]: 10.0.1.1.0

same for folder of migration scripts 😅

Comment (AaronHForgeFlow) [reply purchase_request_procurement/__manifest__.py]: It is already 10.0.1.2.1 :stuck_out_tongue:

Comment (MiquelRForgeFlow) [reply purchase_request_procurement/__manifest__.py]: What do you mean? https://github.com/OCA/purchase-workflow/blob/10.0/purchase_request_procurement/__manifest__.py#L7

Comment (AaronHForgeFlow) [reply purchase_request_procurement/__manifest__.py]: I mean, the migration folder is 10.0.1.2.1, same as manifest.

File: purchase_request/__manifest__.py [hunk 6]
```diff
@@ -6,7 +6,7 @@
     "name": "Purchase Request",
     "author": "Eficent Business and IT Consulting Services S.L., "
               "Odoo Community Association (OCA)",
-    "version": "10.0.1.1.1",
+    "version": "10.0.1.2.1",
```
Comment (MiquelRForgeFlow) [inline purchase_request/__manifest__.py]: change here to `10.0.1.2.0`

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM :1st_place_medal:

File: purchase_request_to_rfq/__manifest__.py [hunk 7]
```diff
@@ -7,7 +7,7 @@
     "author": "Eficent, "
               "Acsone SA/NV,"
               "Odoo Community Association (OCA)",
-    "version": "10.0.1.0.0",
+    "version": "10.0.1.1.0",
     "website": "http://www.eficent.com",
```
Comment (nikul-serpentcs) [inline purchase_request_to_rfq/__manifest__.py]: https://github.com/OCA/purchase-workflow

**Review by nikul-serpentcs — APPROVED** [review]
Minor Change
Code review LGTM  :1st_place_medal:

File: purchase_request_to_rfq/__manifest__.py (hunk 7 above)
Comment (AaronHForgeFlow) [reply purchase_request_to_rfq/__manifest__.py]: Comment attended. Please update review. Thank you!

Comment (nikul-serpentcs) [reply purchase_request_to_rfq/__manifest__.py]: Already Updated review :smile: 

Just rebase and squash commits

**Comment by AaronHForgeFlow** [comment]
ping @hveficent this is merged in v11 and v12

**Comment by HviorForgeFlow** [comment]
@sergio-teruel are you planning to review this again?

File: purchase_request_procurement/models/procurement_order.py [hunk 8]
```diff
@@ -27,35 +27,50 @@ def _prepare_purchase_request_line(self):
...
         :return: False
         """
-        return False
+        domain = (
+            ('state', '=', 'draft'),
+            ('picking_type_id', '=', values['picking_type_id']),
+            ('company_id', '=', values['company_id']),
+        )
+        gpo = self.rule_id.group_propagation_option
+        group_id = (gpo == 'fixed' and self.group_id.id) or \
```
Comment (sergio-teruel) [inline purchase_request_procurement/models/procurement_order.py:61]: If the method is @api.model why do you use self.foo?? I think that is not correct

**Review by sergio-teruel — APPROVED** [review]

**Pull request merged**